smailyforprestashop.php

```diff
@@ -251,6 +259,29 @@ public function getContent()
                 $output .= $this->displayConfirmation($this->l('Abandoned cart settings updated'));
             }
         }
+        // RSS
+        if (Tools::isSubmit('smaily_submit_rss')) {
+            $category_id = Tools::getValue('SMAILY_RSS_CATEGORY_ID');
+            $limit = Tools::getValue('SMAILY_RSS_LIMIT');
+            $sort_by = Tools::getValue('SMAILY_RSS_SORT_BY');
+            $sort_order = Tools::getValue('SMAILY_RSS_SORT_ORDER');
+
+            // Update settings.
+            if ($category_id && $category_id === 'all_products' || is_int(intval($category_id))) {
+                Configuration::updateValue('SMAILY_RSS_CATEGORY_ID', $category_id);
+            }
+            if ($limit >= 1 && is_int($limit)) {
```

Comment:
Same here as with above. `$limit` variable value is not guaranteed to be of type `integer`.

Reply:
Typcasting it now. Also checking if 1 >= `$limit` <= 250